packages/react-sdk/src/components/ConnectModalContent.tsx

```diff
@@ -0,0 +1,221 @@
+  const connectWithAuthProvider = useCallback(
+    async (provider: AuthProviderType) => {
+      try {
+        setIsConnecting(true);
+        setError(null);
+        setProviderType(provider);
+
+        await baseConnect.connect({ provider });
+
+        // Hide modal on successful connection
+        onClose();
+      } catch (err) {
+        setError(err as Error);
+      } finally {
+        setIsConnecting(false);
+        setProviderType(null);
+      }
+    },
+    [baseConnect, onClose],
+  );
+
+  // Connect with injected provider (when extension is installed)
+  const connectWithInjected = useCallback(async () => {
+    if (!sdk) {
+      const err = new Error("SDK not initialized");
+      setError(err);
+      return;
+    }
+
+    try {
+      setIsConnecting(true);
+      setError(null);
+      setProviderType("injected");
+
+      await baseConnect.connect({
+        provider: "injected",
+      });
+
+      // Hide modal on successful connection
+      onClose();
+    } catch (err) {
+      setError(err as Error);
+    } finally {
+      setIsConnecting(false);
+      setProviderType(null);
+    }
+  }, [sdk, baseConnect, onClose]);
+
+  // Connect with deeplink (redirect to Phantom mobile app)
+  const connectWithDeeplink = useCallback(() => {
+    try {
+      setIsConnecting(true);
+      setError(null);
+      setProviderType("deeplink");
+
+      // Generate and redirect to deeplink URL
+      const deeplinkUrl = getDeeplinkToPhantom();
+      window.location.href = deeplinkUrl;
+
+      // This code will likely never be reached due to the redirect
+      onClose();
+    } catch (err) {
+      setError(err as Error);
+      setIsConnecting(false);
+      setProviderType(null);
+    }
+  }, [onClose]);
+
+  const appIconStyle: CSSProperties = appIcon
+    ? {
+        width: "56px",
+        height: "56px",
+        borderRadius: "50%",
+        display: "block",
+        margin: "0 auto 24px",
+        objectFit: "cover" as const,
+      }
+    : {};
+
+  const buttonContainerStyle: CSSProperties = {
+    display: "flex",
+    flexDirection: "column" as const,
+    gap: "12px",
+  };
+
+  const dividerStyle: CSSProperties = {
+    display: "flex",
+    alignItems: "center",
+    margin: "24px 0",
+    ...theme.typography.caption,
+    color: theme.secondary,
+    textTransform: "uppercase" as const,
+  };
+
+  const dividerLineStyle: CSSProperties = {
+    flex: 1,
+    height: "1px",
+    backgroundColor: theme.secondary,
+  };
+
+  const dividerTextStyle: CSSProperties = {
+    padding: "0 12px",
+  };
+
+  const errorStyle: CSSProperties = {
+    backgroundColor: "rgba(220, 53, 69, 0.1)",
+    color: "#ff6b6b",
+    border: "1px solid rgba(220, 53, 69, 0.3)",
+    borderRadius: "8px",
+    padding: "12px",
+    marginBottom: "12px",
+    fontSize: "14px",
+  };
+
+  return (
+    <>
+      {/* App Icon */}
+      {appIcon && <img src={appIcon} alt={appName || "App"} style={appIconStyle} />}
```

Comment:
Nit: this fallback is not clear. We could assign it when destructuring the props.